apps/ledger-live-mobile/src/families/avalanchepchain/DelegationFlow/02-EndDate.tsx

```diff
@@ -0,0 +1,200 @@
+import { accountScreenSelector } from "../../../reducers/accounts";
+import { ScreenName } from "../../../const";
+import { TrackScreen } from "../../../analytics";
+import LText from "../../../components/LText";
+import Button from "../../../components/Button";
+import KeyboardView from "../../../components/KeyboardView";
+import TranslatedError from "../../../components/TranslatedError";
+import { getFirstStatusError } from "../../helpers";
+import { localeSelector } from "../../../reducers/settings";
+import type { StackNavigatorProps } from "../../../components/RootNavigator/types/helpers";
+import { AvalancheDelegationFlowParamList } from "./types";
+
+type Props = StackNavigatorProps<
+  AvalancheDelegationFlowParamList,
+  ScreenName.AvalancheDelegationEndDate
+>;
+
+export default function DelegationEndDate({ navigation, route }: Props) {
+  const { colors } = useTheme();
+  const { account } = useSelector(accountScreenSelector(route));
+  const locale = useSelector(localeSelector);
+
+  invariant(account?.type === "Account", "must be account");
+
+  const bridge = getAccountBridge(account);
+
+  const { transaction, setTransaction, status, bridgePending } =
+    useBridgeTransaction(() => {
+      return {
+        account,
+        transaction: {
+          ...route.params.transaction,
+          amount: new BigNumber(route.params.amount ?? 0),
+          mode: "delegate",
+        },
+      };
+    });
+
+  invariant(transaction, "transaction must be defined");
+
+  const onChange = (_, selectedDate: number) => {
+    setTransaction(
+      bridge.updateTransaction(transaction, {
+        endTime: new BigNumber(selectedDate / 1000),
+      }),
+    );
+  };
+
+  const onContinue = () => {
+    navigation.navigate(ScreenName.AvalancheDelegationValidator, {
+      ...route.params,
+      endTime: transaction.endTime,
+    });
+  };
+
+  const blur = useCallback(() => Keyboard.dismiss(), []);
+
+  const timeError = bridgePending ? null : status.errors.time;
+  const warning = getFirstStatusError(status, "warnings");
+
+  const stakeStartTime = Math.floor(Date.now() / 1000) + FIVE_MINUTES;
+  const unixMinEndDate = stakeStartTime + TWO_WEEKS;
+  const unixMaxEndDate = Math.min(
+    stakeStartTime + YEAR,
+    Number(route.params.validator.endTime),
+  );
+
+  const readableMinEndDate = getReadableDate(unixMinEndDate, locale);
+  const readableMaxEndDate = getReadableDate(unixMaxEndDate, locale);
+
+  return (
+    <>
+      <TrackScreen category="DelegationFlow" name="Amount" />
+      <SafeAreaView
+        style={[styles.root, { backgroundColor: colors.background }]}
+      >
+        <KeyboardView style={styles.container}>
+          <TouchableWithoutFeedback onPress={blur}>
+            <View style={[styles.root, { backgroundColor: colors.background }]}>
+              <View style={styles.wrapper}>
+                <View style={styles.datePickers}>
+                  <DateTimePicker
```

Comment:
I'm just guessing from reading the code but I think we have a max date end right ? is it possible to have this on the configuration of this component ? Like user are less likely to do mistake ?

Reply:
Yup, there's a max end date! It's specified on the `Transaction` object:

https://github.com/LedgerHQ/ledger-live/blob/cdb1c6d8aeb51d2393a9a42d454047c8a448ee60/libs/ledger-live-common/src/families/avalanchepchain/types.ts#L32

Inside of `js-getTransactionStatus.ts`, the code checks to see if the time the user has selected is valid. If the user has selected an invalid time, such as a time past the max end date, then `js-getTransactionStatus` will send an error and that error will be displayed to the user.